### PR TITLE
Feature/reserve block

### PR DESCRIPTION
**Pull request by jessicayu626**

### CODE REVIEW

File: native-filesystem/include/native_filesystem.h [hunk 1]
```diff
@@ -17,8 +17,11 @@ namespace nativefs {
 
 typedef struct {
   uint64_t blockid;
-  uint64_t offset;
-  uint32_t len;
+  uint64_t offset;  // Address in the FS
+  uint32_t len;  // length of the data
+  // in the block
+  uint32_t allocated_size;  // total
+  // allocated size for this block
```
Comment (Antzen) [inline native-filesystem/include/native_filesystem.h:24]: 👍

File: native-filesystem/source/native_filesystem.cc [hunk 2]
```diff
@@ -112,6 +118,7 @@ void NativeFS::flushBlock(int block_index) {
 }
 
 void NativeFS::freeRange(uint64_t start, uint64_t end) {
+  LOG(INFO) << "[freeRange] start: " << start << " end: " << end;
```
Comment (LINKIWI) [inline native-filesystem/source/native_filesystem.cc]: Is this log line critical? It seems to be generating a lot of stdout noise in the build

Comment (jessicayu626) [reply native-filesystem/source/native_filesystem.cc]: No. I will remove it. I was debugging this line.

File: native-filesystem/source/native_filesystem.cc [hunk 3]
```diff
@@ -317,4 +354,44 @@ uint64_t NativeFS::getFreeSpace() {
...
+    freeRange(blockInfo.offset, blockInfo.len);
+
+    // Set the new block info
+    blocks[block_index].allocated_size = new_len;
+    blocks[block_index].len = new_len;
+    blocks[block_index].offset = new_offset;
+  }
+
+  return true;
+}
```
Comment (Antzen) [inline native-filesystem/source/native_filesystem.cc:396]: A rather inefficient way to realloc a block, but I guess this is the best option given the native FS used by RDFS.

**Review by Antzen — APPROVED** [review]
Generally looks good to me. We'll probably want to improve it and debug some existing bugs if we have more time though (some of which are commented on within TODO's)

**Pull request merged**